tests/integration/suite/daprd/workflow/crossapp/security.go

```diff
@@ -0,0 +1,150 @@
+	c.place = placement.New(t)
+	c.sched = scheduler.New(t,
+		scheduler.WithLogLevel("debug"))
+	db := sqlite.New(t,
+		sqlite.WithActorStateStore(true),
+		sqlite.WithMetadata("busyTimeout", "10s"),
+		sqlite.WithMetadata("disableWAL", "true"),
+	)
+
+	app1 := app.New(t)
+	app2 := app.New(t)
+
+	// Create registries for each app and register orchestrator/activity
+	c.registry1 = task.NewTaskRegistry()
+	c.registry2 = task.NewTaskRegistry()
+
+	appID1 := uuid.New().String()
+	appID2 := uuid.New().String()
+
+	c.registry2.AddActivityN("ProcessData", func(ctx task.ActivityContext) (any, error) {
+		var input string
+		if err := ctx.GetInput(&input); err != nil {
+			return nil, fmt.Errorf("failed to get input in app2 activity: %w", err)
+		}
+		return fmt.Sprintf("Processed by app2: %s", input), nil
+	})
+
+	c.registry1.AddOrchestratorN("CrossAppWorkflow", func(ctx *task.OrchestrationContext) (any, error) {
+		var input string
+		if err := ctx.GetInput(&input); err != nil {
+			return nil, fmt.Errorf("failed to get input in app1: %w", err)
+		}
+		var output string
+
+		err := ctx.CallActivity("ProcessData",
+			task.WithActivityInput(input),
+			task.WithAppID(appID2)).
+			Await(&output)
+		if err != nil {
+			return nil, fmt.Errorf("failed to execute activity in app2: %w", err)
+		}
+		return output, nil
+	})
+
+	c.daprd1 = daprd.New(t,
+		daprd.WithInMemoryActorStateStore("mystore"),
+		daprd.WithPlacementAddresses(c.place.Address()),
+		daprd.WithScheduler(c.sched),
+		daprd.WithAppID(appID1),
+		daprd.WithAppPort(app1.Port()),
+		daprd.WithLogLevel("debug"),
+	)
+	c.daprd2 = daprd.New(t,
+		daprd.WithInMemoryActorStateStore("mystore"),
+		daprd.WithPlacementAddresses(c.place.Address()),
+		daprd.WithScheduler(c.sched),
+		daprd.WithAppID(appID2),
+		daprd.WithAppPort(app2.Port()),
+		daprd.WithLogLevel("debug"),
+	)
+
+	return []framework.Option{
+		framework.WithProcesses(c.place, c.sched, db, app1, app2, c.daprd1, c.daprd2),
+	}
+}
+
+func (c *security) Run(t *testing.T, ctx context.Context) {
+	c.sched.WaitUntilRunning(t, ctx)
+	c.place.WaitUntilRunning(t, ctx)
+	c.daprd1.WaitUntilRunning(t, ctx)
+	c.daprd2.WaitUntilRunning(t, ctx)
+
+	// Start workflow listeners for each app
+	client1 := client.NewTaskHubGrpcClient(c.daprd1.GRPCConn(t, ctx), backend.DefaultLogger())
+	client2 := client.NewTaskHubGrpcClient(c.daprd2.GRPCConn(t, ctx), backend.DefaultLogger())
+
+	err := client1.StartWorkItemListener(ctx, c.registry1)
+	assert.NoError(t, err)
+	err = client2.StartWorkItemListener(ctx, c.registry2)
+	assert.NoError(t, err)
+
+	id, err := client1.ScheduleNewOrchestration(ctx, "CrossAppWorkflow", api.WithInput("Hello from app1"))
+	require.NoError(t, err)
+
+	_, err = client1.WaitForOrchestrationStart(ctx, id)
+	require.NoError(t, err)
+
+	// this is key, and should fail
+	_, err = client2.FetchOrchestrationMetadata(ctx, id, api.WithFetchPayloads(true))
+	require.Error(t, err, "no such instance exists", "security breach, this should err out")
```

Comment:
I guess thie is validating `FetchOrchestrationMetadata` is scoped to a single app ID? This doesn't belong in the `crossapps` package.

Please add a test showing that calling an activity of an app ID in another _namespace_ doesn't work.

Reply:
added a new `tests/integration/suite/daprd/workflow/security/appscoping.go` and `tests/integration/suite/daprd/workflow/crossapp/crossnamespace.go`